brightStars_DM_sprint_Feb2020/BeyondRadius.ipynb

```diff
@@ -0,0 +1,422 @@
+    "        filter_lc = 'g'\n",
+    "    elif chosenFilter=='HSC-R':\n",
+    "        filter_lc = 'r'\n",
+    "    elif chosenFilter=='HSC-I':\n",
+    "        filter_lc = 'i'\n",
+    "    elif chosenFilter=='HSC-z':\n",
+    "        filter_lc = 'z'\n",
+    "    elif chosenFilter=='HSC-y':\n",
+    "        filter_lc = 'y'\n",
+    "    print(filter_lc)\n",
+    "    mask_path = \"/datasets/hsc/BrightObjectMasks/GouldingMasksS18A/new_S18Amask_{}.reg\".format(\n",
+    "                                            filter_lc)\n",
+    "else:\n",
+    "    mask_path = \"/datasets/hsc/BrightObjectMasks/ArcturusMasks/\" "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If plots are to be saved, make sure the folder exists:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if savePlots:\n",
+    "    if not os.path.isdir('plots/'):\n",
+    "        os.makedirs('plots/')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run\n",
+    "Warning: running cell below can take a while."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "densities, brightMags, brightRads, brightCenters = [], [], [], []\n",
+    "for tract,patches in zip(tractList,patchList):\n",
+    "    # get sky map and WCS for this tract\n",
+    "    skymap = butler.get(\"deepCoadd_skyMap\", tract=tract, filter=chosenFilter)\n",
+    "    tractInfo = skymap[tract]\n",
+    "    wcs = tractInfo.getWcs()\n",
+    "    \n",
+    "    # convert annuli parameters to pixel\n",
+    "    pixScale = geom.radToArcsec(wcs.getPixelScale())\n",
+    "    annSizePix = annSize / pixScale\n",
+    "    \n",
+    "    # load detection catalog for each patch\n",
+    "    if verbose:\n",
+    "        print(\"Loading and appending all detections...\")\n",
+    "    if not src_loaded:\n",
+    "        src = butler.get('deepCoadd_meas', tract=tract, patch=patches[0], \n",
+    "                         filter=chosenFilter)\n",
+    "        src = src[src['detect_isPatchInner']]\n",
+    "        for patch in patches[1:]:\n",
+    "            patchSrc = butler.get('deepCoadd_meas', tract=tract, patch=patch, \n",
+    "                                 filter=chosenFilter)\n",
+    "            patchSrc = patchSrc[patchSrc['detect_isPatchInner']]\n",
+    "            src.extend(patchSrc)\n",
+    "            src_loaded = True\n",
+    "    if verbose:\n",
+    "        print(\" > Done.\")\n",
+    "        \n",
+    "    # set up bright object selector if needed\n",
+    "    if reselectBrightObjects:\n",
+    "        if reselectBrightObjects == \"boxAround\":\n",
+    "            if verbose:\n",
+    "                print(\"Computing box boundaries...\")\n",
+    "            # hacky box for a rectangular subset of patches within a tract:\n",
```

Comment:
I might be missing something, but I'm not sure why you're not just using the patch + wcs info for this purpose?

Reply:
I could, and should; I used to do it using with tract + wcs
(https://github.com/lsst-dm/drp-analysis-sprint-feb20/blob/12eab896dc0e8610b11b64d18c5d6258660584e1/brightStars_DM_sprint_Feb2020/RadiusDensity.ipynb)

The only reason I ended up doing this horrendous thing is because I reran on not-quite-a-tract to get a plot, last minute, with no background subtracted. It seemed more straightforward to do this than handle the concatenation of several patches.

I'll keep this as is for now though; there is still the weak excuse that you might want to make a selection of bright stars that is more specific than just using the boundaries of your tract or patches of interest.